Include/internal/pycore_ceval.h

```diff
@@ -271,7 +271,7 @@ PyAPI_FUNC(PyObject **) _PyObjectArray_FromStackRefArray(_PyStackRef *input, Py_
 PyAPI_FUNC(void) _PyObjectArray_Free(PyObject **array, PyObject **scratch);
 
 PyAPI_FUNC(PyObject *) _PyEval_GetANext(PyObject *aiter);
-PyAPI_FUNC(PyObject *) _PyEval_LoadGlobal(PyObject *globals, PyObject *builtins, PyObject *name);
+PyAPI_FUNC(void) _PyEval_LoadGlobalStackRef(PyObject *globals, PyObject *builtins, PyObject *name, _PyStackRef *writeto);
```

Comment:
```suggestion
PyAPI_FUNC(_PyStackRef) _PyEval_LoadGlobalStackRef(PyObject *globals, PyObject *builtins, PyObject *name);
```

Having a `void` function that returns its result through an argument pointer prevents the compiler from putting values in registers. What's the reason?

Reply:
If you create a deferred stack reference, it needs to be stored (visibly) on the interpreter's stack before any escaping calls are made. Within bytecodes.c, this is enforced by the code generator, but external functions need a bit more work.

It's not strictly necessary for `LOAD_GLOBAL` because we don't make any escaping calls after creating the stack reference, but it will be for `LOAD_ATTR` because of the complexity with checking both descriptors and dictionary attributes.

Reply:
What's the motivation for doing this?
(I assume it's performance. Converting from a `PyObject *` counted reference to a deferred stack reference in the interpreter is safe, but slow?)

As I said, `LOAD_GLOBAL` is irrelevant for performance.

Reply:
1) Creating a (possibly deferred) stack reference instead of `PyObject*` is for performance
2) Passing a pointer to the stack reference instead of returning the `_PyStackRef` by value is for safety

As I wrote above in https://github.com/python/cpython/pull/123128#discussion_r1723489686, `LOAD_GLOBAL` is currently relevant for performance in the free-threaded build.